src/Serilog.Sinks.AzureDataExplorer/Extensions/AzureDataExplorerSinkOptionsExtensions.cs

```diff
@@ -0,0 +1,74 @@
+            return GetKcsbWithAuthentication(kcsb, options);
+        }
+
+        public static KustoConnectionStringBuilder GetKustoEngineConnectionStringBuilder(
+            this AzureDataExplorerSinkOptions options)
+        {
+            var kcsb = new KustoConnectionStringBuilder(GetClusterUrl(options.IngestionEndpointUri),
+                options.DatabaseName);
+            return GetKcsbWithAuthentication(kcsb, options);
+        }
+
+        private static KustoConnectionStringBuilder GetKcsbWithAuthentication(KustoConnectionStringBuilder kcsb,
+            AzureDataExplorerSinkOptions options)
+        {
+            switch (options.AuthenticationMode)
+            {
+                case AuthenticationMode.AadApplicationCertificate:
+                    kcsb = kcsb.WithAadApplicationCertificateAuthentication(options.ApplicationClientId,
+                        options.ApplicationCertificate, options.Authority, options.SendX5C, options.AzureRegion);
+                    break;
+                case AuthenticationMode.AadApplicationKey:
+                    kcsb = kcsb.WithAadApplicationKeyAuthentication(options.ApplicationClientId, options.ApplicationKey,
+                        options.Authority);
+                    break;
+                case AuthenticationMode.AadApplicationSubjectName:
+                    kcsb = kcsb.WithAadApplicationSubjectNameAuthentication(options.ApplicationClientId,
+                        options.ApplicationCertificateSubjectDistinguishedName, options.Authority, options.AzureRegion);
+                    break;
+                case AuthenticationMode.AadApplicationThumbprint:
+                    kcsb = kcsb.WithAadApplicationThumbprintAuthentication(options.ApplicationClientId,
+                        options.ApplicationCertificateThumbprint, options.Authority);
+                    break;
+                case AuthenticationMode.AadApplicationToken:
+                    kcsb = kcsb.WithAadApplicationTokenAuthentication(options.ApplicationToken);
+                    break;
+                case AuthenticationMode.AadAzureTokenCredentials:
+                    kcsb = kcsb.WithAadAzureTokenCredentialsAuthentication(options.TokenCredential);
+                    break;
+                case AuthenticationMode.AadUserToken:
+                    kcsb = kcsb.WithAadUserTokenAuthentication(options.UserToken);
+                    break;
+                case AuthenticationMode.AadManagedIdentity:
+                    kcsb = kcsb.WithAadUserManagedIdentity(options.ApplicationClientId);
+                    break;
+                case AuthenticationMode.AadUserPrompt:
+                default:
+                    kcsb = kcsb.WithAadUserPromptAuthentication(options.Authority);
+                    break;
+            }
+
+            kcsb.ApplicationNameForTracing = "Serilog.Sink.AzureDataExplorer";
```

Comment:
We can discuss and finetune this. May be app version and other things that need to be added. may be even write one and ask in our channel with Asaf/Yochai so that we use the API in the same spirit in which it was written ?